client/extensions/woocommerce/woocommerce-services/views/shipping-label/label-purchase-modal/customs-step/item-row.js

```diff
@@ -0,0 +1,111 @@
+ */
+import React from 'react';
+import PropTypes from 'prop-types';
+import { connect } from 'react-redux';
+import { localize } from 'i18n-calypso';
+
+/**
+ * Internal dependencies
+ */
+import TextField from 'woocommerce/woocommerce-services/components/text-field';
+import {
+	setCustomsItemDescription,
+	setCustomsItemTariffNumber,
+	setCustomsItemOriginCountry,
+} from 'woocommerce/woocommerce-services/state/shipping-label/actions';
+import {
+	getShippingLabel,
+	isLoaded,
+	getFormErrors,
+} from 'woocommerce/woocommerce-services/state/shipping-label/selectors';
+import CountryDropdown from 'woocommerce/woocommerce-services/components/country-dropdown';
+
+const TariffCodeTitle = localize( ( { translate } ) =>
+	<span>{ translate( 'Tariff Code' ) } (<a href="https://hts.usitc.gov/"
+		target="_blank" rel="noopener noreferrer">{ translate( 'look up' ) }</a>)</span>
+);
+
+const ItemRow = ( props ) => {
+	const {
+		errors,
+		packageId,
+		productId,
+		translate,
+		description,
+		defaultDescription,
+		tariffNumber,
+		originCountry,
+		countriesData,
+	} = props;
+
+	return <div className="customs-step__item-row">
+		<TextField
+			id={ packageId + '_' + productId + '_description' }
+			className="customs-step__item-description-column"
+			title={ translate( 'Description' ) }
+			value={ description }
+			placeholder={ defaultDescription }
+			updateValue={ props.setCustomsItemDescription }
+			error={ errors.description } />
+		<CountryDropdown
+			id={ packageId + '_' + productId + '_originCountry' }
+			className="customs-step__item-country-column"
+			title={ translate( 'Origin country' ) }
+			value={ originCountry }
+			updateValue={ props.setCustomsItemOriginCountry }
+			countriesData={ countriesData } />
+		<TextField
+			id={ packageId + '_' + productId + '_tariffNumber' }
+			className="customs-step__item-code-column"
+			title={ <TariffCodeTitle /> }
+			value={ tariffNumber }
+			updateValue={ props.setCustomsItemTariffNumber }
+			error={ errors.tariffNumber } />
+	</div>;
+};
+
+ItemRow.propTypes = {
+	siteId: PropTypes.number.isRequired,
+	orderId: PropTypes.number.isRequired,
+	packageId: PropTypes.string.isRequired,
+	productId: PropTypes.number.isRequired,
+	description: PropTypes.string.isRequired,
+	defaultDescription: PropTypes.string.isRequired,
+	tariffNumber: PropTypes.string.isRequired,
+	originCountry: PropTypes.string.isRequired,
+	errors: PropTypes.object,
+	countriesData: PropTypes.object.isRequired,
+	setCustomsItemDescription: PropTypes.func.isRequired,
+	setCustomsItemTariffNumber: PropTypes.func.isRequired,
+	setCustomsItemOriginCountry: PropTypes.func.isRequired,
+};
+
+const mapStateToProps = ( state, { orderId, siteId, productId } ) => {
+	const loaded = isLoaded( state, orderId, siteId );
+	const shippingLabel = getShippingLabel( state, orderId, siteId );
+	const storeOptions = loaded ? shippingLabel.storeOptions : {};
+
+	return {
+		...shippingLabel.form.customs.items[ productId ],
```

Comment:
`getShippingLabel()` can return `null` and in that case, we'd get a TypeError error. What's the best way to account for that?

Also, can we explicitly write out `productId` as the property name? It was not easily scannable.

Reply:
Using the spread operator is also introducing props that we don't need in this component - `box_id`, `height`, `id`, `isUserSpecifiedWeight`, `length`, `weight`, `width`. Can we map only the two we need, `items` and `itn`?

Reply:
> getShippingLabel() can return null (#) and in that case, we'd get a TypeError error. What's the best way to account for that?

If the shipping labels state tree is not loaded, the `Label Purchase` modal won't be rendered at all, so at this point, in this component we can assume it exists and it's fully loaded.

> Using the spread operator is also introducing props that we don't need in this component - box_id, height, id, isUserSpecifiedWeight, length, weight, width. Can we map only the two we need, items and itn?

We need all of the properties of `form.customs.items[productId]`: `description`, `defaultDescription`, `tariffNumber`, and `originCountry`. But in spirit of readability, I've updated it in https://github.com/Automattic/wp-calypso/pull/24974/commits/474a53aa33745b927fa538772de502004a1dbdae